tests/phpunit/tests/cron/setCronArray.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+/**
+ * Test the `_set_cron_array()` function.
+ *
+ * @group cron
+ * @covers ::_set_cron_array
+ */
+class Tests_Cron_setCronArray extends WP_UnitTestCase {
+
+	public function set_up() {
+		parent::set_up();
+		// Make sure the schedule is clear.
+		_set_cron_array( array() );
+	}
+
+	public function tear_down() {
+		// Make sure the schedule is clear.
+		_set_cron_array( array() );
+		parent::tear_down();
+	}
+
+	/**
+	 * Tests the input validation for the `_set_cron_array()` function.
+	 *
+	 * Includes verifying that invalid input - typically `false` - does not result in a PHP
+	 * deprecation warning on PHP 8.1 or higher.
+	 *
+	 * The warning that we should not see:
+	 * `Deprecated: Automatic conversion of false to array is deprecated`.
+	 *
+	 * @ticket 53635
+	 *
+	 * @dataProvider data_set_cron_array_input_validation
+	 *
+	 * @param mixed $input    Cron "array".
+	 * @param array $expected Expected array entry count of the cron option after update.
+	 */
+	public function test_set_cron_array_input_validation( $input, $expected ) {
+		delete_option( 'cron' );
+		$this->assertTrue( _set_cron_array( $input ) );
+
+		$crons = get_option( 'cron' );
+		$this->assertIsArray( $crons, 'Cron option is not an array.' );
+		$this->assertArrayHasKey( 'version', $crons, 'Cron option does not have a "version" key.' );
+		$this->assertCount( $expected, $crons, 'Cron option does not contain the expected nr of entries.' );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array
+	 */
+	public function data_set_cron_array_input_validation() {
+		return array(
+			'null'        => array(
+				'input'    => null,
+				'expected' => 1,
+			),
+			// Function _get_cron_array() may return `false`, so this is the PHP 8.1 "problem" test.
+			'false'       => array(
+				'input'    => false,
+				'expected' => 1,
+			),
+			'empty array' => array(
+				'input'    => array(),
+				'expected' => 1,
+			),
+			'cron array'  => array(
+				'input'    => array(
+					'version' => 2,
```

Comment:
It's not a big issue but I am unclear why you are including the version in the data passed to `_set_cron_array()`? 

It's not expected as the the function simply overwrites the passed value or appends it to the option if it's not included. 

Question applies throughout PR.

Reply:
This data set is testing the scenario where the version was previously set.